File: api/BeSureApi/Models/InvoiceReconciliation.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel;

namespace BeSureApi.Models
{
    public class InvoiceReconciliationList
    {
        public int Id { get; set; }
        public decimal NetInvoiceAmount{ get; set; }
        public decimal CollectedAmount { get; set; }
        public decimal OutstandingAmount { get; set; }
        public decimal TdsDeductedAmount { get; set; }
        public decimal TdsPaidAmount { get; set; }
        public decimal GstTdsDeductedAmount { get; set; }
        public decimal GstTdsPaidAmount { get; set; }
        public decimal PenaltyAmount { get; set; }
        public decimal SecurityDepositAmount { get; set; }
        public decimal CustomerExpenseAmount { get; set; }
        public decimal OtherDeductionAmount { get; set; }
        public string? OtherDeductionRemarks { get; set; }
        public decimal WriteOffAmount { get; set; }
        public string? WriteOffAmountRemarks { get; set; }
        public int PrimaryAccountableId { get; set; }
        public int SecondaryAccountableId { get; set; }
        public string InvoiceNumber { get; set; }
        public int CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public int? ModifiedBy { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public string CustomerName { get; set; }
    }

    public class InvoiceReconciliationTdsUpload
    {
        public IFormFile CollectionFile { get; set; }
    }
}

File: ui/src/types/invoiceReconciliation.ts
import { array, boolean, Decoder, nullable, number, object, string } from 'decoders';

export interface InvoiceReconciliationListDetail {
Id: number;
NetInvoiceAmount: number;
CollectedAmount:number;
OutstandingAmount:number;
TdsDeductedAmount:number;
TdsPaidAmount:number;
GstTdsDeductedAmount:number;
GstTdsPaidAmount:number;
OtherDeductionAmount:number;
CustomerExpenseAmount:number;
SecurityDepositAmount:number;
PenaltyAmount:number;
WriteOffAmount:number;
InvoiceNumber:string;
CustomerName:string; 
}

export const invoiceReconciliationListDetailDecoder: Decoder<InvoiceReconciliationListDetail> = object({
Id: number,
NetInvoiceAmount:number,
CollectedAmount:number,
OutstandingAmount:number,
TdsDeductedAmount:number,
TdsPaidAmount:number,
GstTdsDeductedAmount:number,
GstTdsPaidAmount:number,
OtherDeductionAmount:number,
CustomerExpenseAmount:number,
SecurityDepositAmount:number,
PenaltyAmount:number,
WriteOffAmount:number,
InvoiceNumber:string,
CustomerName:string,
});

export interface InvoiceReconciliationList {
  InvoiceReconciliationList: InvoiceReconciliationListDetail[];
  TotalRows: number;
  PerPage:number;
}

export const invoiceReconciliationListDecoder: Decoder<InvoiceReconciliationList> = object({
  InvoiceReconciliationList: array(invoiceReconciliationListDetailDecoder),
  TotalRows: number,
  PerPage:number
}); 


export interface InvoiceReconciliationTaxFileUpload {
  CollectionFile: File | null;
  TaxType:string|null;
}
export interface InvoiceReconciliationTaxUploadResult {
IsCollectionUploaded: Boolean;
}

export const invoiceReconciliationTaxUploadResultDecoder: Decoder<InvoiceReconciliationTaxUploadResult> = object({
IsCollectionUploaded: boolean,
});